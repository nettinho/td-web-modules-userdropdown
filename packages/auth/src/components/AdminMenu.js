import React from "react";
import { compose } from "redux";
import PropTypes from "prop-types";
import { Icon, Menu } from "semantic-ui-react";
import { FormattedMessage } from "react-intl";
import { withRouter } from "react-router-dom";
import { ActiveRoute } from "@truedat/core/components/ActiveRoute";
import UserMenuItem from "../users/components/UserMenuItem";
import RolesMenuItem from "../roles/components/RolesMenuItem";
import roleRoutes from "../roles/routes";
import userRoutes from "../users/routes";
import groupRoutes from "../groups/routes";
import AdminRole from "./AdminRole";
export const AdminMenu = ({ location }) => (
  <Menu.Item
    header
    name="administration"
    active={
      ActiveRoute(location, roleRoutes.ROLES) ||
      ActiveRoute(location, userRoutes.USER_LIST) ||
      ActiveRoute(location, groupRoutes.GROUP_LIST)
    }
  >
    <Icon name="setting" />
    <FormattedMessage id="navigation.admin" />
    <Menu.Menu>
      <UserMenuItem />
      <RolesMenuItem />
    </Menu.Menu>
  </Menu.Item>
);

AdminMenu.propTypes = {
  location: PropTypes.object
};

export default compose(
  withRouter,
  AdminRole
)(AdminMenu);
