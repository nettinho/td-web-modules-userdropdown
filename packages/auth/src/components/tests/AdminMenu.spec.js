import React from "react";
import { shallowWithIntl } from "intl-stub";
import { AdminMenu } from "../AdminMenu";

describe("<AdminMenu />", () => {
  it("matches the latest snapshot", () => {
    const location = { pathname: "/" };
    const props = { location };
    const wrapper = shallowWithIntl(<AdminMenu {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
