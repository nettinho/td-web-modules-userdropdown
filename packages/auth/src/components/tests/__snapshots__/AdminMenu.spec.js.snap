// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminMenu /> matches the latest snapshot 1`] = `
<MenuItem
  active={false}
  header={true}
  name="administration"
>
  <Icon
    as="i"
    name="setting"
  />
  <FormattedMessage
    id="navigation.admin"
    values={Object {}}
  />
  <MenuMenu>
    <withRouter(Connect(UserMenuItem)) />
    <withRouter(RolesMenuItem) />
  </MenuMenu>
</MenuItem>
`;
